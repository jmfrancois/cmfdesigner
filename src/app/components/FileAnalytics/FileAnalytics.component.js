import React from 'react';
import PropTypes from 'prop-types';

function getPath(path, cwd) {
	if (path && cwd) {
		return path.replace(cwd, '$');
	}
	return path;
}

function FileAnalytics(props) {
	return (
		<li key={props.path}>
			{getPath(props.path, props.cwd)}
			<ul>
				{props.components.map((component, index) => (
					<li key={index}>
						{component.name}
						<ul>
							<li>Type: {component.type}</li>
							{component.propTypes && <li>has PropTypes</li>}
							{component.displayName && <li>displayName: {component.displayName}</li>}
						</ul>
						{!component.displayName && (
							<div className="alert alert-danger">
								<p>No DisplayName: In the context of CMF it s a requirement to add
									displayName to your component
								</p>
							</div>
						)}
						{!component.propTypes && (
							<div className="alert alert-warning">
								<p>No PropTypes: It's always better to add propTypes</p>
							</div>
						)}
					</li>
				))}
			</ul>
		</li>
	);
}

FileAnalytics.propTypes = {
	path: PropTypes.string,
	cwd: PropTypes.string,
	components: PropTypes.arrayOf(PropTypes.object),
};
FileAnalytics.defaultProps = {
	components: [],
};

export default FileAnalytics;
