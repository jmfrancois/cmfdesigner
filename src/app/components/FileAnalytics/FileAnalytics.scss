.title {
    font-size: medium;
    color: $dark-silver;
    display: inline-block;
}

.opn {
    display: inline-block;
    margin-left: $padding-normal;
}
.content {
    margin-left: $padding-normal;
    padding-left: $padding-normal;
    border-left: 1px solid $dark-silver;

    h3 {
        font-size: medium;
        :global(.tc-svg-icon) {
            width: $svg-xs-size;
            height: $svg-xs-size;
        }
    }
}
