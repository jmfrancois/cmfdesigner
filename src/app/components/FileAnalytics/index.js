import FileAnalytics from './FileAnalytics.component';

export default FileAnalytics;
