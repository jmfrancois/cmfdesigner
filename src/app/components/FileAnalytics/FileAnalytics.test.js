import React from 'react';
import { shallow } from 'enzyme';

import FileAnalytics from './FileAnalytics.component';

describe('FileAnalytics', () => {
	it('should render', () => {
		const wrapper = shallow(
			<FileAnalytics />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
