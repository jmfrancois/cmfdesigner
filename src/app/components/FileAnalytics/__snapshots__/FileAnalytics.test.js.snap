// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileAnalytics should render 1`] = `
<li>
  <ul>
    
  </ul>
</li>
`;
